src/python/gudhi/barycenter.py

```diff
@@ -0,0 +1,342 @@
+import ot
+import numpy as np
+import scipy.spatial.distance as sc
+
+
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Theo Lacombe
+#
+# Copyright (C) 2019 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+
+def _proj_on_diag(w):
+    '''
+        Util function to project a point on the diag.
+    '''
+    return np.array([(w[0] + w[1])/2 , (w[0] + w[1])/2])
+
+
+def _proj_on_diag_array(X):
+    '''
+    :param X: (n x 2) array encoding the points of a persistent diagram.
+    :returns: (n x 2) array encoding the (respective orthogonal) projections of the points onto the diagonal
+    '''
+    Z = (X[:,0] + X[:,1]) / 2.
+    return np.array([Z , Z]).T
```

Comment:
`Z = X.mean(axis=1)` and `return Z.reshape(-1,1).repeat(2,axis=1)` would be less readable, ok.

Reply:
I guess the Z = X.mean(axis=1) is as readable as what I have done (for the second part, I would think `np.array([Z, Z]).T` is more readable, but I didn't investigate if it has any impact in terms of performances (this is however not the bottleneck of the algorithm anyway).